### ISSUE TITLE
White Screen when exiting video playback on Raspberry Pi

### ISSUE THREAD

**Issue by zachron** [MythTV/mythtv#421]
* **Platform**: Raspbian OS, Linux raspberrypi 5.10.63-v8+ #1459 SMP PREEMPT Wed Oct 6 16:42:49 BST 2021 aarch64 GNU/Linux

* **MythTV version**: fixes/31 [v31.0-167-g25f1bb1d12]

* **Package version**: N/A built from source

* **Component**: frontend

### What steps will reproduce the bug?

after watching a video (but not a recording) i get a white screen that the only way to get it to go away is to restart mythfrontend.

### How often does it reproduce? Is there a required condition?
it happens every time when i finish watching a video. 

### What is the expected behaviour?
to go back to the menu system.

### Additional information

this happens on both 32 and 64 bit raspberry pi's. (i just currently have it running on a 64 bit pi). It seems very similar to https://code.mythtv.org/trac/ticket/13544 (but is not this, cause its already been fixed in the version i am using) when the white screen is showing, the menu (i assume behind it) is still active, and if i push the select buttons on the remote, i can get the same video (or another video) to play, and video does show up, and then goes back to the white screen when the video ends.

I have attached a debug log of when this happens to this ticket. 

If you need any other information, please let me know.

[mythfrontend.20211202004620.717.log](https://github.com/MythTV/mythtv/files/7638770/mythfrontend.20211202004620.717.log)

**Comment by ulmus-scott** [comment]
You appear to be using the Terra theme.  Does your issue also occur with MythCenter-wide?

**Comment by zachron** [comment]
I am going to assume its the same issue, but its a black screen with MythCenter-wide.

**Comment by ulmus-scott** [comment]
Could you try master to see if it is an issue there?  As I understand it, there have been fairly extensive changes to the UI and playback code.

Also, when did you start noticing this issue?  After a mythtv update?  Other system updates?

**Comment by zachron** [comment]
> Could you try master to see if it is an issue there? As I understand it, there have been fairly extensive changes to the UI and playback code.

I am not really ready to update by backend and frontends to v32 quite yet. 

> Also, when did you start noticing this issue? After a mythtv update? Other system updates?

it started happening about 3 weeks ago, i had updated raspberry pi os about a week to 2 weeks before it started happening, so i am unsure if it is related to that. i have since in trying to resolve the issue, started with a fresh install of raspberry pi os, and installed everything fresh.

**Comment by ulmus-scott** [comment]
> I am not really ready to update by backend and frontends to v32 quite yet.

Sorry, I meant test separately from your existing setup.  If you can build a combined frontend/backend on a Raspberry Pi, it wont be the fastest backend, but it should suffice for testing, especially since this is a frontend issue and all it needs to do is serve a video to the frontend.  Obviously, you would need a video file that the new, separate master branch backend can access.

**Comment by zachron** [comment]
> Sorry, I meant test separately from your existing setup. If you can build a combined frontend/backend on a Raspberry Pi, it won't be the fastest backend, but it should suffice for testing, especially since this is a frontend issue and all it needs to do is serve a video to the frontend. Obviously, you would need a video file that the new, separate master branch backend can access.

yeah i can do that, ill reply back here after i get that all setup

**Comment by zachron** [comment]
> Sorry, I meant test separately from your existing setup. If you can build a combined frontend/backend on a Raspberry Pi, it won't be the fastest backend, but it should suffice for testing, especially since this is a frontend issue and all it needs to do is serve a video to the frontend. Obviously, you would need a video file that the new, separate master branch backend can access.

I just finished some testing of this setup - and the issue does not appear to be present in v32 (or at least i could not replicate it). and unrelated, i would say video playback is generally much improved with v32

**Comment by zachron** [comment]
so.. i believe i have this kind of resolved - so messing with v32, i noticed that the MMAL decoder was gone from mythtv (which is what i had been using in v31), and i thought i would take a look at what is currently being used in v31, so i went into my settings, and while it was set to the MMAL profile, that was actually using V4L2 Acceleration (and MMAL is not listed as an option) so i went to see what other options were available. V4L2 Acceleration (decode only) was the only other option (besides standard), so i thought i would try that. That does not have the issue i am experiencing. to be sure i switched it back to just V4L2 Acceleration and tried it again, and sure enough, white screen. V4L2 Acceleration (decode only) no white screen, and back to menu.   so it appears there is some kind of issue with the V4L2 Acceleration, but the same issue does not exist for the V4L2 Acceleration (decode only)
